rtos/Mutex.h

```diff
@@ -89,7 +106,8 @@ class Mutex : private mbed::NonCopyable<Mutex> {
 
       @note You cannot call this function from ISR context.
      */
-    osStatus lock(uint32_t millisec=osWaitForever);
+    MBED_DEPRECATED_SINCE("mbed-os-5.10.0", "Replaced with try_lock(millisec) function")
+    osStatus lock(uint32_t millisec);
```

Comment:
Now you've added the error to the implementation, osErrorParameter and osErrorISR are now impossible. I think that's fine, but remove them from doxygen and have a similar note here that RTOS errors other than the above are treated as fatal.

Reply:
Adapted the return values and added missing note.